signer/tests/integration/block_observer.rs

```diff
@@ -1415,3 +1420,249 @@ async fn block_observer_updates_dkg_shares_after_observing_bitcoin_block() {
+    faucet.generate_blocks(1);
+
+    // Okay now there is a deposit, and it has been confirmed. We should
+    // pick it up automatically.
+    let signal = signal_receiver.recv();
+    let Ok(SignerSignal::Event(SignerEvent::BitcoinBlockObserved)) = signal.await else {
+        panic!("Not the right signal")
+    };
+
+    // We should have two donations if we processed both blocks correctly
+    let donations = fetch_output(&db, TxOutputType::Donation).await;
+    assert_eq!(donations.len(), 2);
+
+    let TxOutput { txid, output_index, amount, .. } = donations[0];
+    assert_eq!(amount, donation_amount);
+    assert_eq!(output_index, donation_outpoint.vout);
+    assert_eq!(txid.deref(), &donation_outpoint.txid);
+
+    let TxOutput { txid, output_index, amount, .. } = donations[1];
+    assert_eq!(amount, donation_amount_2);
+    assert_eq!(output_index, donation_outpoint_2.vout);
+    assert_eq!(txid.deref(), &donation_outpoint_2.txid);
+
+    // But we should have no deposits
+    assert!(fetch_input(&db, TxPrevoutType::Deposit).await.is_empty());
+
+    // Since the deposit check is buried in the logs, we also manually check
+    // that the deposit request validation in the block observer failed as
+    // expected.
+    let request = CreateDepositRequest {
+        outpoint: deposit_request.outpoint.clone(),
+        reclaim_script: deposit_request.reclaim_script.clone(),
+        deposit_script: deposit_request.deposit_script.clone(),
+    };
+    let bitcoin_client = ctx.get_bitcoin_client();
+    let validate_result =
+        signer::block_observer::DepositRequestValidator::validate(&request, &bitcoin_client, false);
+    match validate_result.await {
+        Err(Error::BitcoinTxCoinbase(tx)) if tx == deposit_request.outpoint.txid => {}
+        _ => panic!("Expected a err, got something else"),
+    }
+
+    testing::storage::drop_db(db).await;
+}
+
+fn make_coinbase_deposit_request(
+    rpc: &bitcoincore_rpc::Client,
+    max_fee: u64,
+    signers_public_key: bitcoin::XOnlyPublicKey,
+) -> (bitcoin::Transaction, signer::bitcoin::utxo::DepositRequest) {
+    let deposit_inputs = DepositScriptInputs {
+        signers_public_key,
+        max_fee,
+        recipient: PrincipalData::from(StacksAddress::burn_address(false)),
+    };
+    let reclaim_inputs = ReclaimScriptInputs::try_new(50, bitcoin::ScriptBuf::new()).unwrap();
+
+    let deposit_script = deposit_inputs.deposit_script();
+    let reclaim_script = reclaim_inputs.reclaim_script();
+
+    let script_pub_key =
+        sbtc::deposits::to_script_pubkey(deposit_script.clone(), reclaim_script.clone());
+    let address = Address::from_script(&script_pub_key, bitcoin::Network::Regtest).unwrap();
+
+    let block_hash = rpc.generate_to_address(1, &address).unwrap()[0];
+    let transactions = rpc.get_block(&block_hash).unwrap();
+
+    let deposit_tx = transactions.txdata[0].clone();
+    // Sanity check
+    assert!(deposit_tx.input[0].previous_output.is_null());
```

Comment:
Maybe use the `.is_coinbase()` here since that's the check done in the observer?

Reply:
Oh right, updated in [c4ea8af](https://github.com/stacks-sbtc/sbtc/pull/1592/commits/c4ea8af172192f4b09216d56fba5e1f44e3673d9)